main.cpp

```diff
@@ -18,13 +18,19 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& _rewriter;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter):_rewriter(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        if (const CStyleCastExpr *Expr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast")) {
+            	_rewriter.InsertText(Expr->getLParenLoc(), "static_cast");
+            	_rewriter.ReplaceText(Expr->getLParenLoc(), 1, "<");
+            	_rewriter.ReplaceText(Expr->getRParenLoc(), 1, ">(");
+            	auto tmp_expr = Lexer::getLocForEndOfToken(Expr->getEndLoc(), 0, *Result.SourceManager, Result.Context->getLangOpts());

```

Comment:
Please describe the reason of using the`Lexer::getLocForEndOfToken()` function here.
What would be the difference if `rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")");` was used here?
What would be the difference if  `rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")");` was used here?

Reply:
`Lexer::getLocForEndOfToken()` - Computes the source location just past the end of the token at this source location. 
`rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")")` - Inserts the text at the position shifted by one after the end of token.We might run into a problem if the name of the variable behind the token  is more than one character.
`rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")")` - I think if we use this, then there will be no difference

Reply:
`styleCastExpr->getEndLoc()` returns the end of the node location.